runtime/src/chpl-mem.c

```diff
@@ -37,6 +36,10 @@ void chpl_mem_init(void) {
 
 
 void chpl_mem_exit(void) {
+  extern void chpl_privatization_exit(void);
+  extern void chpl_qsbr_exit(void);
+  chpl_qsbr_exit();
+  chpl_privatization_exit();
```

Comment:
It's probably more reasonable to put these calls in chpl_exit_common just before chpl_mem_exit. Also, the style of putting the 'extern' decl in the function body is good practice in Chapel code, but in C code we like to have these declared in the relevant header.